### PR TITLE
Feature/nanoflow

### PR DESCRIPTION
**Pull request by jseremba**
* Add nanoflow support on click event

### CODE REVIEW

File: src/Carousel.webmodeler.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class preview extends Component<CarouselContainerProps, {}> {
     }
 
     private getImages(props: CarouselContainerProps): Image[] {
-        const defaultImages = [ { url: require("./img/Preview.png") } ];
+        const defaultImages = [ { url: require("./img/Preview.png"), openPageLocation: props.openPageLocation } ];
```
Comment (djkwagala) [inline src/Carousel.webmodeler.ts]:
````
const defaultImages = [ {
    url: require("./img/Preview.png"),
    openPageLocation: props.openPageLocation
} ];

if (props.dataSource === "static") {

````

File: src/Carousel.xml [hunk 2]
```diff
@@ -41,6 +41,16 @@
                     <category>On click action</category>
                     <description>Page to open when the image is clicked</description>
                 </property>
+                <property key="openPageLocation" type="enumeration" defaultValue="content">
```
Comment (djkwagala) [inline src/Carousel.xml]: I recommend renaming the key from `openPageLocation` to `displayPageAs` or `openPageAs`

File: src/Carousel.xml [hunk 3]
```diff
@@ -95,5 +105,15 @@
             <category>Behavior</category>
             <description>The page to open when an image is clicked</description>
         </property>
+        <property key="openPageLocation" type="enumeration" defaultValue="content">
```
Comment (djkwagala) [inline src/Carousel.xml]: I recommend renaming the key from `openPageLocation` to `displayPageAs` or `openPageAs`

File: src/components/Carousel.ts [hunk 4]
```diff
@@ -47,6 +48,7 @@ interface CustomEvent extends Event {
 }
 
 type Direction = "right" | "left";
+export type PageLocations = "popup" | "modal" | "content";
```
Comment (djkwagala) [inline src/components/Carousel.ts]: start with the default one, and rename to PageLocation
`export type PageLocation = "content" | "popup" | "modal";`

File: src/components/CarouselContainer.ts [hunk 5]
```diff
@@ -254,7 +255,8 @@ export default class CarouselContainer extends Component<CarouselContainerProps,
                 context,
                 error: error => window.mx.ui.error(
                     `An error occurred while opening form ${image.onClickForm} : ${error.message}`
-                )
+                ),
+                location: this.props.dataSource === "static" ? image.openPageLocation : this.props.openPageLocation
```
Comment (djkwagala) [inline src/components/CarouselContainer.ts]: Please use this standard format.
```
location: this.props.dataSource === "static" 
    ? image.openPageLocation
    : this.props.openPageLocation
```

**Review by djkwagala — CHANGES_REQUESTED** [review]

File: src/Carousel.webmodeler.ts [hunk 6]
```diff
@@ -35,26 +35,15 @@ export function getPreviewCss() {
...
-        visibilityMap.onClickForm = false;
-    } else if (props.onClickOptions === "callMicroflow") {
-        visibilityMap.onClickForm = false;
-    } else if (props.onClickOptions === "showPage") {
-        visibilityMap.onClickMicroflow = false;
-    }
+    visibilityMap.onClickForm = props.onClickOptions === "showPage";
+    visibilityMap.onClickMicroflow = props.onClickOptions === "callMicroflow";
+    visibilityMap.onClickNanoflow = props.onClickOptions === "callNanoflow";
+    visibilityMap.openPageAs = props.onClickOptions === "showPage";
```
Comment (Andries-Smit) [inline src/Carousel.webmodeler.ts]: should be placed after line 43, (as it belongs to the clickform) and the order of the XML

File: src/Carousel.xml (hunk 2 above)
Comment (jseremba) [reply src/Carousel.xml]: Chosen `openPageAs`

File: src/components/CarouselContainer.ts [hunk 7]
```diff
@@ -232,19 +235,30 @@ export default class CarouselContainer extends Component<CarouselContainerProps,
...
+            });
         } else if (image.onClickForm) {
             window.mx.ui.openForm(image.onClickForm, {
                 context,
                 error: error => window.mx.ui.error(
                     `An error occurred while opening form ${image.onClickForm} : ${error.message}`
-                )
+                ),
+                location: this.props.dataSource === "static"
+                    ? image.openPageAs
```
Comment (Andries-Smit) [inline src/components/CarouselContainer.ts]: not sure if this correct? we dont do this for `image.onClickForm` either. Please check with Edwin for the logic

**Pull request merged**